R/req-perform-stream.R

```diff
@@ -35,13 +44,27 @@ req_perform_stream <- function(req, callback, timeout_sec = Inf, buffer_kb = 64)
   withr::defer(close(stream))
 
   continue <- TRUE
+  incomplete <- TRUE
+  buf <- raw()
   while(continue && isIncomplete(stream) && Sys.time() < stop_time) {
-    buf <- readBin(stream, raw(), buffer_kb * 1024)
+    buf <- c(buf, readBin(stream, raw(), buffer_kb * 1024))
```

Comment:
This update looks great!

Reply:
I like how it rolls back to the original `while(continue && isIncomplete(stream) && Sys.time() < stop_time)` loop